### PR TITLE
Webhooks integration

### PR DESCRIPTION
**Pull request by harshal0815**
Winter of code PR# 1

### CODE REVIEW

File: lib/puptime/service/base.rb [hunk 1]
```diff
@@ -56,6 +56,28 @@ def raise_error_level
         @error_level += 1 if @error_level < 3
       end
 
+      def run
+        @scheduler_job_id = @scheduler.every @interval, overlap: false, job: true do
```
Comment (abhishekkumar2718) [inline lib/puptime/service/base.rb]: Wait, what? What does this block of code even do?

File: lib/puptime/service/base.rb [hunk 2]
```diff
@@ -56,6 +56,28 @@ def raise_error_level
...
+          Puptime::Service::Redis.ping
+        end
+      end
+
+      # Changes to be done to add all error notifications and pass it as an argument
+      def notifier_base(service_name)
+        Puptime::NotificationQueue.enqueue_notification(service_name)
+      end
+
+      def self.error_level
```
Comment (abhishekkumar2718) [inline lib/puptime/service/base.rb:59]: What's wrong with class constant?

File: lib/puptime/service/tcp.rb [hunk 3]
```diff
@@ -24,18 +23,12 @@ def initialize(name, id, type, interval, options = {})
...
       end
 
-      def run
-        @scheduler_job_id = @scheduler.every @interval, overlap: false, job: true do
-          ping
-        end
-      end
-
-      def ping
+      def self.ping
```
Comment (abhishekkumar2718) [inline lib/puptime/service/tcp.rb]: I don't understand why this has to be a class method. Likewise for other ping functions

**Review by abhishekkumar2718 — CHANGES_REQUESTED** [review]

**Comment by z0marlin** [comment]
What is this PR supposed to do? Please add a decent description to your PRs and commits

**Pull request closed without merging**